src/dbnode/encoding/proto/common.go

```diff
@@ -76,11 +76,17 @@ const (
 	// Single bit op codes that get encoded into the compressed stream and
 	// inform the iterator / decoder how it should interpret subsequent
 	// bits.
-	opCodeNoMoreDataOrTimeUnitChange = 0
-	opCodeMoreData                   = 1
+	opCodeNoMoreDataOrTimeUnitChangeAndOrSchemaChange = 0
+	opCodeMoreData                                    = 1
 
-	opCodeNoMoreData     = 0
-	opCodeTimeUnitChange = 1
+	opCodeNoMoreData                      = 0
+	opCodeTimeUnitChangeAndOrSchemaChange = 1
```

Comment:
Hm how does this work?  Do you now always encode two bits if the More Data bit is set?

Reply:
One bit if there is just another write, two bits if its the end of the stream, and 4 bits in the case where there is a time unit change or schema change. The README has a good diagram